File: src/main/old/bbarena/model/event/game/FameEvent.java
package bbarena.model.event.game;

import bbarena.model.Arena;

public class FameEvent extends GameEvent {

    private Integer fameCoach0 = 0;
    private Integer fameCoach1 = 0;

    public FameEvent() {
    }

    public void setFameCoach0(int fameCoach0) {
        this.fameCoach0 = fameCoach0;
    }

    public void setFameCoach1(int fameCoach1) {
        this.fameCoach1 = fameCoach1;
    }

    @Override
    protected void doEvent(Arena arena) {
        arena.getScoreBoard(0).setFame(fameCoach0);
        arena.getScoreBoard(1).setFame(fameCoach1);
    }

    @Override
    protected void undoEvent(Arena arena) {
        arena.getScoreBoard(0).setFame(0);
        arena.getScoreBoard(1).setFame(0);
    }

    @Override
    public String getString() {
        StringBuilder res = new StringBuilder();
        res.append(getClass().getSimpleName())
                .append("[fame:")
                .append(fameCoach0)
                .append("]");
        return res.toString();
    }
}

File: src/main/kotlin/bbarena/model/event/game/FameEvent.kt
package bbarena.model.event.game

import bbarena.model.Arena

class FameEvent(private val fameByCoach: IntArray) : GameEvent() {

    val oldFameByCoach = mutableListOf<Int>()

    override val string: String
        get() = {
            val res = StringBuilder()
            res.append(javaClass.simpleName)
                    .append("[")
            fameByCoach.forEachIndexed { coach, fame ->
                res.append("fame")
                        .append(coach)
                        .append(": ")
                        .append(fame)
                        .append(", ") }
            res.append("]")
            res.toString()
        }.invoke()

    fun setFameCoach(coach: Int, fame: Int) {
        fameByCoach[coach] = fame
    }

    override fun doEvent(arena: Arena) {
        fameByCoach.forEachIndexed { coach, fame ->
            oldFameByCoach[coach] = arena.getScoreBoard(coach).fame!!
            arena.getScoreBoard(coach).fame = fame
        }
    }

    override fun undoEvent(arena: Arena) {
        fameByCoach
    }
}
